test_util/test_runner.py

```diff
@@ -120,37 +130,53 @@ def integration_test(
         add_env: a python dict with any number of key=value assignments to be passed to
             the test environment
     """
-    log.info('Transfering integration_test.py')
     test_script = pkg_filename('integration_test.py')
-    tunnel.remote_cmd(['mkdir', '-p', test_dir])
-    tunnel.write_to_remote(test_script, test_dir+'/integration_test.py')
+    pytest_docker = pkg_filename('docker/py.test/Dockerfile')
 
-    test_container_name = 'int_test_' + str(int(time.time()))
     dns_search = 'true' if test_dns_search else 'false'
-    test_cmd = [
-        'docker', 'run', '-v', test_dir+'/integration_test.py:/integration_test.py',
-        '--net=host', '--name='+test_container_name]
-    test_cmd.extend([
-        '-e', 'DCOS_DNS_ADDRESS=http://'+dcos_dns,
-        '-e', 'MASTER_HOSTS='+','.join(master_list),
-        '-e', 'PUBLIC_MASTER_HOSTS='+','.join(master_list),
-        '-e', 'SLAVE_HOSTS='+','.join(agent_list),
-        '-e', 'PUBLIC_SLAVE_HOSTS='+','.join(public_agent_list),
-        '-e', 'REGISTRY_HOST=127.0.0.1',
-        '-e', 'DCOS_VARIANT='+variant,
-        '-e', 'DNS_SEARCH='+dns_search,
-        '-e', 'AWS_ACCESS_KEY_ID='+aws_access_key_id,
-        '-e', 'AWS_SECRET_ACCESS_KEY='+aws_secret_access_key,
-        '-e', 'AWS_REGION='+region])
+    test_env = [
```

Comment:
We only use BASH_TEMPLATE once, and we only use it for this stuff. Why not make all these formattings part of it. Something like:

```

"""
#!/bin/bash
set -euo pipefail
set -x
export DCOS_DNS_ADDRESS=http://{dcos_dns}
export MASTER_HOSTS={master_hosts}
export PUBLIC_MASTER_HOSTS={master_hosts}
export SLAVE_HOSTS={slave_hosts}
# ...

{test_cmd}
""".format(
  dcos_dns=dcos_dns,
  master_hosts=','.join(master_list),
  slave_hosts=','.join(agent_list)
)
```

(Still leaving it as the template up above, .format down below). Then we're doing one less layer of string formatting (It all just happens at once into cmd_script, and we should be fine living just with python format strings, unless there's some particular reason you want to use gen.template, which is also fine to use).